proxy/src/telemetry/metrics/latency.rs

```diff
@@ -146,25 +140,19 @@ where
 
 }
 
-
 impl<'a> IntoIterator for &'a Histogram {
-    type Item = &'a u32;
-    type IntoIter = slice::Iter<'a, u32>;
+    type Item = u32;
+    type IntoIter = iter::Map<
+        slice::Iter<'a, prometheus::Counter>,
+        fn(&'a prometheus::Counter) -> u32
+    >;
 
     fn into_iter(self) -> Self::IntoIter {
-        self.buckets.iter()
+        self.buckets.iter().map(|&a| { let a: u64 = a.into(); a as u32 })
```

Comment:
The truncation to u32 should be moved to the metrics API code that is about to be deleted, instead of being here. Otherwise we'd likely forget to fix this when the Metrics API code is deleted. Also we should add a big ugly comment to this conversion in that code saying that it's wrong and we're doing it as a temporary workaround until that API is deleted.

Reply:
Done in 013e728833c7299726da465394a1705429f4cfc8.